draft-hendrickson-privacypass-public-metadata-issuance.md

```diff
@@ -43,10 +46,220 @@ TODO Abstract
+  described in {{public-issuer-configuration}}.
+- Challenge value: `challenge`, an opaque byte string. For example, this might
+  be provided by the redemption protocol in [AUTHSCHEME].
+- Metadata value: `metadata`, an opaque byte string.
+
+Given this configuration and these inputs, the two messages exchanged in
+this protocol are described below. The constant `Nk` is defined as token type 0x1234 in
+{{PROTOCOL}}.
+
+## Client-to-Issuer Request {#public-request}
+
+The Client first creates an issuance request message for a random value
+`nonce` using the input challenge and Issuer key identifier as follows:
+
+~~~
+nonce = random(32)
+challenge_digest = SHA256(challenge)
+token_input = concat(0x1234, // Token type field is 2 bytes long
+                     nonce,
+                     challenge_digest,
+                     token_key_id)
+blinded_msg, blind_inv =
+  Blind(pkI, metadata, PrepareIdentity(token_input))
+~~~
+
+The PrepareIdentity and Blind functions are defined in
+{{PBLINDRSA}} and {{PBLINDRSA}}, respectively.
+The Client stores the nonce, challenge_digest, and metadata values locally for use
+when finalizing the issuance protocol to produce a token (as described
+in {{public-finalize}}).
+
+The Client then creates a TokenRequest structured as follows:
+
+~~~
+struct {
+  uint16_t token_type = 0x1234; /* Type Public Metadata Blind RSA (2048-bit) */
+  uint8_t truncated_token_key_id;
+  opaque metadata<1..2^16-1>;
+  uint8_t blinded_msg[Nk];
+} TokenRequest;
+~~~
+
+The structure fields are defined as follows:
+
+- "token_type" is a 2-octet integer, which matches the type in the challenge.
+
+- "truncated_token_key_id" is the least significant byte of the `token_key_id`
+  ({{public-issuer-configuration}}) in network byte order (in other words, the
+  last 8 bits of `token_key_id`).
+
+- "metadata" is the opaque metadata value that all `blinded_msg` values are encoded for.
+
+- "blinded_msg" is the Nk-octet request defined above.
+
+The Client then generates an HTTP POST request to send to the Issuer Request
+URI, with the TokenRequest as the content. The media type for this request
+is "application/private-token-request". An example request is shown below:
+
+~~~
+:method = POST
+:scheme = https
+:authority = issuer.example.net
+:path = /request
+accept = application/private-token-response
+cache-control = no-cache, no-store
+content-type = application/private-token-request
+content-length = <Length of TokenRequest>
+
+<Bytes containing the TokenRequest>
+~~~
+
+## Issuer-to-Client Response {#public-response}
+
+Upon receipt of the request, the Issuer validates the following conditions:
+
+- The TokenRequest contains a supported token_type.
+- The TokenRequest.truncated_token_key_id corresponds to the truncated key
+  ID of an Issuer Public Key.
+- The TokenRequest.blinded_msg is of the correct size.
+- The TokenREquest.metadata satisfies issuer requirements.
```

Comment:
I would drop "satisfies issuer requirements" since it's not clear what that means. I would instead lean on text elsewhere that says something like "metadata values are defined by the deployment and have privacy considerations; see <architecture doc> for more details."

Reply:
We still should have some text saying that the issuer/attester can drop the token request if it has invalid values (where invalid values are defined by the deployment). Otherwise its not clear what parties have an option of denying a TokenRequest.

Reply:
@smhendrickson can we do something with this text?